parsing/cLexer.ml

```diff
@@ -754,9 +750,9 @@ let token_text = function
   | (con, "") -> con
   | (con, prm) -> con ^ " \"" ^ prm ^ "\""
 
-let func next_token cs =
+let func next_token ?loc cs =
   let loct = loct_create () in
-  let cur_loc = ref (Loc.create !current_file 1 0 0 0) in
+  let cur_loc = ref (Option.default Loc.(create ToplevelInput 1 0 0 0) loc) in
```

Comment:
I suggest you factor the creation of the `1 0 0 0` location to a function in `Loc`.

Reply:
done